### PR TITLE
bundle import, take first available existing entity if duplicates

### PR DESCRIPTION
**Pull request by ereteog**
> Close #https://github.com/threatgrid/iroh/issues/3880

In `bundle/import` route, when duplicate entities are actually found, use the last provided entity to avoid further duplicate and notify it in the response about these duplicates.

<a name="qa">[§](#qa)</a> QA
============================

- use `bundle/import` route without the the `external-key-prefixes` value: `test` to import the following bundle:
[SWE Bundle.json.zip](https://github.com/threatgrid/ctia/files/4847617/SWE.Bundle.json.zip)
For all imported entities, the field `result` should have the value `created`.
- make the same request once again:
For all imported entities, the field `result` should have the value `created`.
- import the same bundle another time but with the `external-key-prefixes` value: `swe-alarm`:
For all imported entities, the field `result` should have the value `existing` with no error.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: enables CTIA `bundle/import` to deal with duplicated external ids.
```

### CODE REVIEW

File: src/ctia/bundle/core.clj [hunk 1]
```diff
@@ -166,23 +166,18 @@
                              first
                              :entity
                              with-long-id
-                             ent/un-store)
-          num-old-entities (count old-entities)]
+                             ent/un-store)]
+      (when (< (count old-entities) 1)
```
Comment (vemv) [inline src/ctia/bundle/core.clj:170]: `(< (count old-entities) 1)` means `(zero? (count old-entities))`, which I'm not sure is the right thing?

File: test/ctia/bundle/core_test.clj [hunk 2]
```diff
@@ -44,3 +48,33 @@
             :query "source_ref:*malware*"}
            (sut/relationships-filters "id" {:source_type :malware
                                             :related_to [:source_ref]})))))
+
+(deftest with-existing-entity-test
+  (testing "with-existing-entity"
+    (let [indicator-id-1 (make-id "indicator")
+          indicator-id-2 (make-id "indicator")
+          indicator-id-3 (make-id "indicator")]
+      (testing "1 existing external id"
```
Comment (vemv) [inline test/ctia/bundle/core_test.clj:57]: You could try an `are` here

As I see it, it's not only about DRYing, but also about making the differences evident (rn I'd have to squint and look for differences very carefully)

File: src/ctia/bundle/core.clj (hunk 1 above)
Comment (ereteog) [reply src/ctia/bundle/core.clj:170]: That's a bug, thx!
I am punished for not testing that the logging is actually done :-).

File: test/ctia/bundle/core_test.clj (hunk 2 above)
Comment (ereteog) [reply test/ctia/bundle/core_test.clj]: Tried to address this with a test fn, it's a little bit "harder" to make it readable with logging test. Tell me what you think.

**Comment by ereteog** [comment]
@vemv FYI the force push is just a clean of the test ns and rebase (removed unused fn import)

**Comment by ereteog** [comment]
> Close #https://github.com/threatgrid/iroh/issues/3880

In `bundle/import` route, when duplicate entities are actually found, use the last provided entity to avoid further duplicate and notify it in the response about these duplicates.

**Comment by hshahman** [comment]
Verified in rel-v1.52

**Pull request merged**